ecs/README.md

```diff
@@ -7,121 +7,174 @@ Full instructions can be found in the
 [Getting Started: Running in ECS](https://linkerd.io/getting-started/ecs/)
 Guide.
 
+[Amazon ECS](https://aws.amazon.com/ecs/) is a container management service.
+This guide will demonstrate routing and monitoring your services using Linkerd
+in ECS.
+
+## Overview
+
+This guide will demonstrate setting up Linkerd as a service mesh, Consul for
+service discovery, a hello-world sample app, and linkerd-viz for monitoring, all
+on a fresh ECS cluster.
+
+The following components make up the system:
+
+* `ECS`: Docker container management. Every ECS instance runs the following
+  Docker containers:
+  * `linkerd`: proxies requests to `hello-world`
+  * `consul-agent`: local service discovery agent
+  * [`consul-registrator`](https://github.com/gliderlabs/registrator): bridge
+  between Docker and Consul, automatically registers services with consul
+* `hello-world`: example ECS task deployed separately from foundational
+  `ECS`+`linkerd`+`consul-agent` configuration, composed of `hello`, `world`,
+  and `world-v2` services
+* [`linkerd-viz`](https://github.com/linkerd/linkerd-viz): ECS task deployed
+  separately from foundational `ECS`+`linkerd`+`consul-agent` configuration,
+  provides a monitoring dashboard for all service traffic
+* `consul-server`: service discovery back-end, runs on a single EC2 instance
+
 ## Initial Setup
 
-```bash
-KEY_PAIR=<MY KEY PAIR NAME>
-```
+First, launch a cluster of EC2 instances orchestrated by ECS. You can use a
+CloudFormation template to create a VPC, security group, an autoscaling group
+for launching EC2 instances configured to connect to ECS. You can download or
+clone the [linkerd-examples repo](https://github.com/linkerd/linkerd-examples/tree/master/ecs)
+to get the template.
 
-## Security Group
+Open the CloudFormation console and deploy the `linkerd-ecs-cluster.yml` template
+or if you have the AWS CLI installed and configured run:
 
-```bash
-GROUP_ID=$(aws ec2 create-security-group --group-name l5d-demo-sg --description "Linkerd Demo" | jq -r .GroupId)
-aws ec2 authorize-security-group-ingress --group-id $GROUP_ID \
-  --ip-permissions \
-  FromPort=22,IpProtocol=tcp,ToPort=22,IpRanges=[{CidrIp="0.0.0.0/0"}] \
-  FromPort=4140,IpProtocol=tcp,ToPort=4140,IpRanges=[{CidrIp="0.0.0.0/0"}] \
-  FromPort=9990,IpProtocol=tcp,ToPort=9990,IpRanges=[{CidrIp="0.0.0.0/0"}] \
-  FromPort=3000,IpProtocol=tcp,ToPort=3000,IpRanges=[{CidrIp="0.0.0.0/0"}] \
-  FromPort=8500,IpProtocol=tcp,ToPort=8500,IpRanges=[{CidrIp="0.0.0.0/0"}] \
-  IpProtocol=-1,UserIdGroupPairs=[{GroupId=$GROUP_ID}]
 ```
-
-## `consul-server`
-
-```bash
-aws ec2 run-instances --image-id ami-62e0d802 \
-  --instance-type m4.xlarge \
-  --user-data file://consul-server-user-data.txt \
-  --placement AvailabilityZone=us-west-1a \
-  --tag-specifications "ResourceType=instance,Tags=[{Key=Name,Value=l5d-demo-consul-server}]" \
-  --key-name $KEY_PAIR --security-group-ids $GROUP_ID
+aws cloudformation deploy \
```

Comment:
mind keeping `KEY_PAIR=<MY KEY PAIR NAME>`? that way the `--parameter-overrides KeyName...` (and others farther down) are more easily copy/paste-able.